### DISCUSSION TITLE
cannot load autoloader. what is wrong?

### DISCUSSION THREAD

**Discussion by kenvanhuffel** [mpdf/mpdf#1733] [Q&A]
Hi
I did use composer to generate the mpdf lib.
I dropped the vendor in my project and used the autoload.php to laod in classes.
I use php 7.4 and i get this error:
What am I doing wrong????

Warning
: require(C:\xampp\htdocs\admnet\libs\mPDF\vendor\composer/../../src/functions-dev.php): failed to open stream: No such file or directory in
C:\xampp\htdocs\admnet\libs\mPDF\vendor\composer\autoload_real.php
on line
55

**Comment by finwe** [comment]
"dropped the vendor"? Did you use composer? If not, just use composer directly, you'll save yourself a lot of headaches.

Judging by the paths, you "dropped" a wrong directory structure. See above.

**Reply by kenvanhuffel** [reply]
> "dropped the vendor"? Did you use composer? If not, just use composer directly, you'll save yourself a lot of headaches.
> 
> Judging by the paths, you "dropped" a wrong directory structure. See above.

I first did composer install. then it created a vendor directory.
Then I copied this vendor directory to my project and I did include the autoload wich is the correct path.
C:\xampp\htdocs\admnet\libs\mPDF\vendor\ is the right root directory.
I load: C:\xampp\htdocs\admnet\libs\mPDF\vendor\autoload.php wich exists.
What am I doing wrong?

**Reply by finwe** [reply]
you have to run `composer require mpdf/mpdf` directly in your project directory. otherwise the generated paths will be wrong. the vendor directory will be created directly by this command.

if you have to _copy_ the vendor directory, you're doing it wrong.

**Reply by kenvanhuffel** [reply]
Ow, I see. But how can I do this on my localhost and then use it online in production? I dont want to use composer on my online server. I really hate composer, all the dependencies and it makes it really difficult :(

**Reply by finwe** [reply]
Let me rephrase: if you have to copy **only** the vendor directory, you're doing it wrong.

Of course, copying your entire project along with its dedicated vendor directory is completely OK.

I believe you'll find that composer actually makes things simpler if you give it a chance. But not running it on production server is a legitimate course of action, then just copy the result of local install.

**Reply by kenvanhuffel** [reply]
Thanks. but the localhost runs: localhost/projectname while online its not a folder. its just domain.com
I also dont understand the versions. I did run "composer require mpdf/mpdf" and I only get a composer.json.
I also dont know what to do next and how I can install a lower version of fpdi? and also how i can require tcpdf into the project.
I just dont udnerstand anything about it. I watched 3 hours composer stuff and dont understand anything about it :(

**Reply by finwe** [reply]
Subfolder or main domain makes no difference in using composer. only the relative paths from your files to the vendor/autoload.php file matter.

When you run `composer require mpdf/mpdf`, the highest possible version of mPDF based on your PHP version will be installed. Its version constraint can be seen in composer.json file.

Why would you install lower version of fpdi? 

You can simply require tcpdf by running `composer require tecnickcom/tcpdf` in your project, you then would have mPDF and TCPDF alongside each other in one project (though I don't understand the need for both of them).
